src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,183 @@
+            String rating = results.getString("rating");
+            String signatureDish = results.getString("sigDish");
+            String healthRating = results.getString("healthRating").toUpperCase();
+            restaurants.add(new Restaurant(name, rating, signatureDish,healthRating));
+        }
+        return restaurants;
+    }
+    public static boolean thereAreDuplicates(Connection conn, String name) throws SQLException{
+        boolean same = false;
+        ArrayList<Restaurant> restaurants= new ArrayList<>();
+        Statement stmt = conn.createStatement();
+        ResultSet results=stmt.executeQuery("SELECT restaurantname FROM restaurants");
+        while (results.next()) {
+            if (results.getString("restaurantname").equalsIgnoreCase(name)){
+                same=true;
+            }
+        }
+        return same;
+    }
+
+    public static void main(String[] args) throws SQLException {
+        Server.createWebServer().start();
+        Connection conn = DriverManager.getConnection("jdbc:h2:./main");
+        Statement stmt = conn.createStatement();
+        stmt.execute("CREATE TABLE IF NOT EXISTS restaurants (id IDENTITY, restaurantname VARCHAR, rating VARCHAR, sigDish VARCHAR, healthRating VARCHAR)");
+        Spark.staticFileLocation("/templates");
+
+        Spark.init();
+
+        Spark.get("/",
+                ((request, response) -> {
+                    HashMap m = new HashMap();
+                    ArrayList<Restaurant> restaurants;
+                    restaurants=selectRestaurants(conn);
+//                    for (Restaurant looping : restaurants) {
+//                        if (looping.getName() == replyIdNum) {
+//                            threads.put(message);
+//                        }
+//                    }
+                    Session session = request.session();
+                    String userName = session.attribute("userName");
+
+                    m.put("restaurant", restaurants);
+                    m.put("userName", userName);
+                    return new ModelAndView(m,"home.html");
+                }),new MustacheTemplateEngine()
+                );
+        Spark.post("/create-restaurant",
+                ((request, response) -> {
+                    String name=request.queryParams("name").toUpperCase();
+                    String rating=request.queryParams("rating");
+                    String signatureDish=request.queryParams("sigDish").toLowerCase();
+                    String healthGrade=request.queryParams("healthGrade").toUpperCase();
+                    if(!rating.contains("*")){
+                        response.redirect("/");
+                        return "";
+                    }
+                    if(healthGrade.length()!=1){
+                        response.redirect("/");
+                        return "";
+                    }
+//                    ArrayList<Restaurant> restaurants;
```

Comment:
Alvaro, really well written.  Clean and concise.  I like the checks for bad data.